### ISSUE TITLE
Using `==` in `format!` causes mismatched-arg-count

### ISSUE THREAD

**Issue by vallentin** [rust-analyzer/rust-analyzer#7085]
As the title says, using `==` in `format!` and other related macros, causes ``Expected 2 arguments, found 1 rust-analyzer(mismatched-arg-count)``.

This only seems to apply to `==` and not `!=`, `>=`, `<=`, `+`, `-`, etc.

```rust
let (a, b) = (1, 2);

println!("{}", a == b);
print!("{}", a == b);
format!("{}", a == b);
```

However, wrapping the comparison in parentheses does not trigger the issue:

```rust
println!("{}", (a == b));
```

![image](https://user-images.githubusercontent.com/17464404/103323022-25a78f80-4a41-11eb-81ee-a4e83cd3188d.png)

This issue must have been introduced lately, as a few weeks ago this was not an issue.

Issue exists on both stable and the latest nightly release:
- rust-analyzer version: 2020-12-28 (1d53075)
- rust-analyzer version: nightly (77ad203)
